benches/rocksdb_writebatch.rs

```diff
@@ -11,87 +11,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/// This suite contains benchmarks for several different configurations generated
+/// dynamically. Thus it has `harness = false`.
+extern crate criterion;
+
+extern crate rocksdb;
+extern crate tempdir;
+
+use criterion::{black_box, Bencher, Criterion};
 use rocksdb::{Writable, WriteBatch, DB};
 use tempdir::TempDir;
-use test::Bencher;
 
 fn writebatch(db: &DB, round: usize, batch_keys: usize) {
     let v = b"operators are syntactic sugar for calls to methods of built-in traits";
     for r in 0..round {
         let batch = WriteBatch::new();
         for i in 0..batch_keys {
             let k = format!("key_round{}_key{}", r, i);
-            batch.put(k.as_bytes(), v).unwrap();
+            batch.put(black_box(k.as_bytes()), black_box(v)).unwrap();
```

Comment:
Why is `black_box` necessary here?

Reply:
Compiler is very smart and it will inlines constant values. In real world, these values are produced by user and it cannot be inlined, so there would be a performance difference sometimes.